daras_ai_v2/language_model.py

```diff
@@ -819,6 +874,26 @@ def run_language_model(
     ), "Pleave provide exactly one of { prompt, messages }"
 
     model: LargeLanguageModels = LargeLanguageModels[str(model)]
+
+    if model.is_deprecated:
+        if model.redirect_to:
```

Comment:
why are these 2 nested conditions

Reply:
the other way to write would be:
```python
if model.is_deprecated and model.redirect_to:
    ...
elif model.is_deprecated:
  ...
```

Reply:
sounds good but I dont see an elif condition required in your code

Reply:
It is this at the moment:

```python
if model.is_deprecated:
  if model.redirect_to:
    return run_language_model(...)
  else:
    raise UserError("model is deprecated")
```

Yes moving to non-nested `if`-`elif` will be more obvious

Reply:
![image](https://github.com/user-attachments/assets/18cf8601-22fe-409e-a793-ae745ccdea36)

Reply:
ah. that doesn't look right. will fix

Reply:
fyi: I would have dont it like this, but what you have works
```
if model.is_deprecated:
	if not model.redirect_to:
	    raise UserError(f"Model {model} is deprecated.")	
    return run_language_model(...)
```